release-calendar/src/client/components/ReleaseCard.tsx

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../stylesheets/releaseCard.scss';
+import * as React from 'react';
+import {ApiService} from '../api-service';
+import {Channel, Promotion} from '../../types';
+import {channelTitles} from './ChannelTable';
+import moment from 'moment-timezone';
+
+interface History {
+  [key: string]: {title: string; emoji: string; emojiName: string};
+}
+
+export interface ReleaseCardProps {
+  title: string;
+  className: string;
+}
+
+interface ReleaseCardState {
+  releaseDates: Promotion[];
+}
+
+export class ReleaseCard extends React.Component<
+  ReleaseCardProps,
+  ReleaseCardState
+> {
+  private apiService: ApiService;
+
+  constructor(props: Readonly<ReleaseCardProps>) {
+    super(props);
+    this.state = {
+      releaseDates: null,
+    };
+    this.apiService = new ApiService();
+  }
+
+  async componentDidMount(): Promise<void> {
+    const release = await this.apiService.getReleaseDates(this.props.title);
+    this.setState({releaseDates: release.promotions});
+  }
+
+  history: History = {
+    [Channel.NIGHTLY]: {
+      title: channelTitles.find((channel) => channel.channel == Channel.NIGHTLY)
+        .title,
+      emoji: '🌙',
+      emojiName: 'moon',
+    },
+    [Channel.OPT_IN_EXPERIMENTAL]: {
+      title: `${
+        channelTitles.find(
+          (channel) => channel.channel == Channel.OPT_IN_EXPERIMENTAL,
+        ).title
+      }, ${
+        channelTitles.find((channel) => channel.channel == Channel.OPT_IN_BETA)
+          .title
+      }`,
+      emoji: '✋',
+      emojiName: 'hand',
+    },
+    [Channel.PERCENT_EXPERIMENTAL]: {
+      title: `${
+        channelTitles.find(
+          (channel) => channel.channel == Channel.PERCENT_EXPERIMENTAL,
+        ).title
+      }, ${
+        channelTitles.find((channel) => channel.channel == Channel.PERCENT_BETA)
+          .title
+      }`,
+      emoji: '🧪',
+      emojiName: 'experiment',
+    },
```

Comment:
This assumes that a release is promoted to perc-beta at the same time as perc-experimental, but let's treat this as separate channels and have separate lines.

Maybe in the future we'll consolidate these channels into one. But for now let's stay consistent throughout the app.